app/uk.gov.hmrc.helptosavestridefrontend/controllers/StrideController.scala

```diff
@@ -59,7 +59,12 @@ class StrideController @Inject() (val authConnector:       AuthConnector,
 
   private def checkIfAlreadyEnrolled(nino: String)(ifNotEnrolled: ⇒ Future[Result])(implicit request: Request[AnyContent], hc: HeaderCarrier): Future[Result] = { // scalastyle:ignore
       def updateSessionIfEnrolled(enrolmentStatus: EnrolmentStatus)(implicit hc: HeaderCarrier): EitherT[Future, String, Unit] = enrolmentStatus match {
-        case Enrolled    ⇒ keyStoreConnector.put(HtsSession(AlreadyHasAccount)).map(_ ⇒ ())
+        case Enrolled ⇒ {
+          helpToSaveConnector.getNSIUserInfo(nino).map { nsiUserInfo ⇒
+            val x = keyStoreConnector.put(HtsSession(AlreadyHasAccount, nsiUserInfo))
```

Comment:
can we give this a better name  ans handle response of  `put ` ?

Reply:
@Jackie1988 ^^^

Reply:
updated, thanks